File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t square)
{
    return (square>0 && square<65)? 1ull << (square - 1ull) : 0ull;    
}

uint64_t total(void){
    return ~0ull;
}

File: solutions/clojure/grains/1/src/grains.clj
(ns grains)

(def two-pow
  (memoize (fn [n]
    (if (== n 0) (bigint 1)
     (* 2 (two-pow  (dec n))))))
  )

(defn square [n] 
  (two-pow (dec n))
)

(defn total [] 
  (dec (two-pow 64))
)
